miden/tests/integration/exec_iters.rs

```diff
@@ -0,0 +1,155 @@
+        },
+        VmState {
+            clk: 5,
+            stack: [14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0].to_elements(),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 6,
+            stack: [13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0].to_elements(),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 7,
+            stack: [12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0, 0, 0].to_elements(),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 8,
+            stack: [17, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0, 0, 0].to_elements(),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 9,
+            stack: [1, 17, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0].to_elements(),
+            fmp: fmp,
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 10,
+            stack: [17, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0, 0, 0].to_elements(),
+            fmp: Felt::new(1073741825),
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 11,
+            stack: [0, 17, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0, 0, 0].to_elements(),
+            fmp: Felt::new(1073741825),
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 12,
+            stack: [
+                0, 0, 17, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0, 0, 0,
+            ]
+            .to_elements(),
+            fmp: Felt::new(1073741825),
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 13,
+            stack: [
+                0, 0, 0, 17, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 1, 0, 0, 0, 0,
+            ]
+            .to_elements(),
+            fmp: Felt::new(1073741825),
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 14,
+            stack: [
+                0, 0, 0, 0, 17, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 2, 1, 0, 0, 0, 0,
+            ]
+            .to_elements(),
+            fmp: Felt::new(1073741825),
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 15,
+            stack: [
+                0, 0, 0, 0, 0, 17, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, 0, 0, 0,
+            ]
+            .to_elements(),
+            fmp: Felt::new(1073741825),
+            memory: mem.clone(),
+        },
+        VmState {
+            clk: 16,
+            stack: [
+                1073741825, 0, 0, 0, 0, 17, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 1, 0, 0, 0, 0,
+            ]
+            .to_elements(),
+            fmp: Felt::new(1073741825),
+            memory: vec![
+                (1_u64, slice_to_word(&[13, 14, 15, 16])),
+                (1073741825_u64, slice_to_word(&[0, 0, 0, 0])),
+            ],
```

Comment:
Similar issue to before: it seems like memory gets updated a cycle earlier than it should.

Also, surprising that `17` didn't get saved into memory. Though, maybe this is a bug somewhere else?